src/utils/shorten-url.js

```diff
@@ -4,65 +4,66 @@
 
 // @flow
 
-import queryString from 'query-string';
-import url from 'url';
-
 const accessToken = 'b177b00a130faf3ecda6960e8b59fde73e902422';
-export function shortenUrl(urlToShorten: string): Promise<string> {
+export async function shortenUrl(urlToShorten: string): Promise<string> {
   let longUrl = urlToShorten;
   if (!longUrl.startsWith('https://profiler.firefox.com/')) {
-    const parsedUrl = url.parse(longUrl);
-    const parsedUrlOnCanonicalHost = Object.assign({}, parsedUrl, {
-      protocol: 'https:',
-      host: 'profiler.firefox.com',
-    });
-    longUrl = url.format(parsedUrlOnCanonicalHost);
+    const parsedUrl = new URL(longUrl);
+    parsedUrl.protocol = 'https';
+    parsedUrl.host = 'profiler.firefox.com';
+    parsedUrl.port = '';
+    longUrl = parsedUrl.toString();
   }
 
-  const bitlyQueryUrl =
-    'https://api-ssl.bitly.com/v3/shorten?' +
-    queryString.stringify({
-      longUrl,
-      domain: 'perfht.ml',
-      format: 'json',
-      access_token: accessToken,
-    });
+  const bitlyQueryUrl = 'https://api-ssl.bitly.com/v4/shorten';
+  const payload = {
+    long_url: longUrl,
+    domain: 'perfht.ml',
+  };
 
-  return fetch(bitlyQueryUrl)
-    .then(response => response.json())
-    .then(json => json.data.url);
-}
+  const response = await fetch(bitlyQueryUrl, {
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(payload),
+  });
 
-export async function expandUrl(urlToExpand: string): Promise<string> {
-  const bitlyQueryUrl =
-    'https://api-ssl.bitly.com/v3/expand?' +
-    queryString.stringify({
-      shortUrl: urlToExpand,
-      format: 'json',
-      access_token: accessToken,
-    });
-  const response = await fetch(bitlyQueryUrl);
-  const json = await response.json();
-
-  if (!json.data) {
-    // In case of an error, json.data is null.
+  if (!response.ok) {
     throw new Error(
-      `An error happened while expanding the shortened url ${urlToExpand}: ${
-        json.status_txt
-      } (${json.status_code})`
+      `An error happened while shortening the long url ${longUrl}: ${
```

Comment:
Do you know how this error gets surfaced in the UI?

Reply:
actually, I don't remember :) I'll look into it before landing.

Reply:
In case of an error, we put the full url in the "permalink" input, and we output the received error in a console.warn: https://github.com/firefox-devtools/profiler/blob/cf8039636346c65ce8854472aa652320eef9e7ae/src/components/app/MenuButtons/Permalink.js#L49-L54